app/src/main/java/com/github/onlynotesswent/ui/common/Popups.kt

```diff
@@ -153,44 +161,89 @@ fun CreationDialog(
+fun FileSystemPopup(
+    onDismiss: () -> Unit,
+    folderViewModel: FolderViewModel,
+    onMoveHere: (Folder?) -> Unit = {}
+) {
   var selectedFolder by remember { mutableStateOf<Folder?>(folderViewModel.selectedFolder.value) }
   var folderSubFolders by remember { mutableStateOf<List<Folder>>(emptyList()) }
   val userRootFolders = folderViewModel.userRootFolders.collectAsState()
 
+  // Modify the subfolder when selected Folder changes, best way I found how to do it as when done
+  // sequentially it takes a bit of time for the selected Folder to change which causes a bug
+  // where the subfolders don't update. Could fix this problem using a wait but that would depend
+  // on the internet speed of the user.
+  LaunchedEffect(selectedFolder) {
+    if (selectedFolder != null) {
+      folderViewModel.getSubFoldersOfNoStateUpdate(
+          selectedFolder!!.id, onSuccess = { subFolders -> folderSubFolders = subFolders })
+    }
+  }
   Dialog(onDismissRequest = { onDismiss() }) {
     Box(
         modifier =
             Modifier.testTag("FileSystemPopup")
-                .fillMaxWidth(0.9f) // Adjust the popup width
-                .fillMaxHeight(0.5f)
+                .fillMaxWidth(0.95f) // Adjust the popup width
+                .fillMaxHeight(0.7f)
                 .padding(16.dp)
                 .background(
                     color = MaterialTheme.colorScheme.background,
                     shape = RoundedCornerShape(12.dp))) {
-          Column(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
+          Column(modifier = Modifier.fillMaxSize()) {
             Box(
                 modifier =
                     Modifier.fillMaxWidth()
                         .background(MaterialTheme.colorScheme.primary)
                         .padding(vertical = 12.dp, horizontal = 16.dp)) {
-                  Text(
-                      text =
-                          if (selectedFolder == null)
-                              stringResource(R.string.file_system_folders_in_root)
-                          else "Folders in: ${selectedFolder!!.name}",
-                      style = MaterialTheme.typography.bodyMedium,
-                      color = MaterialTheme.colorScheme.onPrimary)
+                  Row(
+                      modifier = Modifier.fillMaxWidth(),
+                      verticalAlignment = Alignment.CenterVertically) {
+                        IconButton(
+                            modifier = Modifier.testTag("goBackFileSystemPopup"),
+                            onClick = {
+                              if (selectedFolder != null) {
+                                if (selectedFolder!!.parentFolderId != null) {
+                                  folderViewModel.getFolderByIdNoStateUpdate(
+                                      selectedFolder!!.parentFolderId!!,
+                                      onSuccess = { parentFolder -> selectedFolder = parentFolder })
+                                } else {
```

Comment:
Is the else needed? since you check whether selecter folder is not null, and if it is null you perform: `selectedFolder = null`

Reply:
Yes it is necessary. As we set selectedFolder to null when selectedFolder!!.parentFolderId is null not when the selectedFolder is null.